rust/src/lib.rs

```diff
@@ -39,26 +40,26 @@ impl InstrumentationContext {
         )
     }
 
-    pub fn enter(&mut self, fi: &FrameInfo) -> Result<()> {
+    pub fn enter(&mut self, func_index : u32, func_name : Option<&String>) -> Result<()> {
```

Comment:
```suggestion
    pub fn enter(&mut self, func_index : u32, func_name : Option<&str>) -> Result<()> {
```

Along the same lines as the auto conversion / type coercion  as the `&Vec` -> `&[u8]` suggestions above.